src/main.rs

```diff
@@ -25,13 +30,28 @@ const BITS_GREY        : [u8; 2] = [0, 1];
 const BITS_DARK_BLUE   : [u8; 2] = [0, 0];
 const BITS_LIGHT_BLUE  : [u8; 2] = [1, 0];
 
-fn list_tiles(input_dir : &Path) -> Result<Paths, String> {
+#[derive(StructOpt, Debug)]
+struct Opt {
+    #[structopt(help = "Path to tiles to insert", parse(from_os_str))]
+    input: PathBuf,
+    #[structopt(help = "Font file to write to", parse(from_os_str))]
+    target: PathBuf,
+    #[structopt(short = "a", long = "append",
+                help = "Append extra data to the end of the file",
+                parse(from_os_str))]
+    append: Option<PathBuf>,
+    #[structopt(short = "c", long = "compress",
+                help = "Compress the generated data using Sega's CMP")]
+    compress: bool,
+}
+
+fn list_tiles(input_dir : &PathBuf) -> Result<Paths, FontCreationError> {
     if !input_dir.exists() {
-        return Err(format!("Directory does not exist: {}", input_dir.to_string_lossy()));
+        return Err(FontCreationError::new(format!("Directory does not exist: {}", input_dir.to_string_lossy())));
     }
     match glob(&input_dir.join("*.png").to_string_lossy()) {
         Ok(glob) => return Ok(glob),
-        Err(e) => return Err(format!("Error listing files: {}", e)),
+        Err(e) => return Err(FontCreationError::new(format!("Error listing files: {}", e))),
     }
```

Comment:
I'd rewrite that like

```rust
glob(&input_dir.join("*.png").to_string_lossy())
    .with_context(|e| format!("Error listing files: {}", e))
```

Reply:
Does this also require using `failure` types? Attempting to use this nets the following error:

```
  --> src/main.rs:52:5
   |
48 |   fn list_tiles(input_dir : &PathBuf) -> Result<Paths, FontCreationError> {
   |                                          -------------------------------- expected `std::result::Result<glob::Paths, sss_fontbuild::errors::FontCreationError>` because of return type
...
52 | /     glob(&input_dir.join("*.png").to_string_lossy())
53 | |         .with_context(|e| format!("Error listing files: {}", e))
   | |________________________________________________________________^ expected struct `sss_fontbuild::errors::FontCreationError`, found struct `quicli::prelude::Context`
   |
   = note: expected type `std::result::Result<_, sss_fontbuild::errors::FontCreationError>`
              found type `std::result::Result<_, quicli::prelude::Context<std::string::String>>`
```

Reply:
Actually, it looks like I'm still seeing a variant on this even when I'm using an error type that implements `Fail`.